hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,570 @@
+        phone_suffix = phone[len(phone)-5:len(phone)]
+
+        # Crear dominios de búsqueda usando patrones con comodines
+        domain = [('email', 'ilike', f'{email_prefix}%'), ('phone', 'ilike', f'%{phone_suffix}')]
+
+        # Realizar la búsqueda con el dominio definido
+        existing_contact = request.env['res.partner'].search(domain, limit=1)
+
+
+        list_not_title = ['lang', 'company_type', 'email', 'website', 'phone']
+
+        for key, value in contact_data.items():
+            if isinstance(value, str) and key not in list_not_title:
+                    contact_data[key] = value.title()  # Convierte la primera letra de cada palabra a mayúscula
+
+        if existing_contact:
+            # Actualizar el contacto existente si se encontró
+            existing_contact.write(contact_data)
+            logger.info("Contacto actualizado con ID %s", existing_contact.id)
+            return {"message": f"Contacto actualizado con ID: {existing_contact.id}.", "contact_id": existing_contact.id}
+        else:
+            # Crear un nuevo contacto si no se encontró ninguno existente
+            if 'email' not in contact_data:
+                contact_data['email'] = email
+            if 'phone' not in contact_data and phone:
+                contact_data['phone'] = phone
+            new_contact = request.env['res.partner'].create(contact_data)
+            logger.info("Nuevo contacto creado con ID %s", new_contact.id)
+            return {"message": f"Nuevo contacto creado con ID: {new_contact.id}.", "contact_id": new_contact.id}
+        
+    @route('/manage_contact_store', methods=['POST'], type='json', auth='user')
+    def manage_contact_store(self):
+        
+        env = request.env
+        data = request.jsonrequest
+        store_name = data.get('store_name')
+        partner_id = data.get('partner_id')
+        contact_data = data.get('contact_data', {})  # Datos del contacto a crear
+
+        # Verificar si ya existe un contacto a partir de los últimos digitos del nombre
+        store_name_suffix = store_name[len(store_name)-4:len(store_name)]
+
+        # Crear dominios de búsqueda usando patrones con comodines
+        domain = [('name', 'ilike', f'%{store_name_suffix}')]
+
+        # Realizar la búsqueda con el dominio definido
+        existing_contact = request.env['res.partner'].search(domain, limit=1)
+
+
+        list_not_title = ['lang', 'company_type', 'email', 'website', 'phone']
+
+        for key, value in contact_data.items():
+            if isinstance(value, str) and key not in list_not_title:
+                    contact_data[key] = value.title()  # Convierte la primera letra de cada palabra a mayúscula
+
+        if existing_contact:
+            # Retornar el ID
+
+            logger.info("Contacto encontrado con ID %s", existing_contact.id)
+            return {"message": f"Contacto encontrado con ID: {existing_contact.id}.", "contact_id": existing_contact.id}
+        else:
```

Comment:
Unnecessary "else" after "return", remove the "else" and de-indent the code inside it